source/dem/velocity_verlet_integrator.cc

```diff
@@ -36,39 +32,53 @@ VelocityVerletIntegrator<dim>::integrate_pre_force(
 template <int dim>
 void
 VelocityVerletIntegrator<dim>::integrate_post_force(
-  Particles::ParticleHandler<dim> &particle_handler,
-  Tensor<1, dim>                   g,
-  double                           dt)
+  Particles::ParticleHandler<dim> &        particle_handler,
+  Tensor<1, dim>                           g,
+  double                                   dt,
+  std::unordered_map<int, Tensor<1, dim>> &momentum,
+  std::unordered_map<int, Tensor<1, dim>> &force,
+  std::unordered_map<int, double> &        MOI,
+  std::unordered_map<int, Tensor<1, dim>> &acceleration)
 {
   for (auto particle = particle_handler.begin();
        particle != particle_handler.end();
        ++particle)
     {
       // Get the total array view to the particle properties once to improve
       // efficiency
-      auto particle_properties = particle->get_properties();
+      unsigned int    particle_id           = particle->get_id();
+      auto            particle_MOI          = MOI[particle_id];
+      auto            particle_properties   = particle->get_properties();
+      Tensor<1, dim> &particle_momentum     = momentum[particle_id];
+      Tensor<1, dim> &particle_force        = force[particle_id];
+      Tensor<1, dim> &particle_acceleration = acceleration[particle_id];
 
       for (int d = 0; d < dim; ++d)
         {
           // Calculate the acceleration
-          particle_properties[PropertiesIndex::acc_x + d] =
-            g[d] + particle_properties[PropertiesIndex::force_x + d] /
-                     particle_properties[PropertiesIndex::mass];
+          particle_acceleration[d] =
+            g[d] +
+            particle_force[d] / particle_properties[PropertiesIndex::mass];
```

Comment:
Quick question. How often do we divide by mass compared to multiply by it? it might be a good idea to store 1/mass instead if over the entire code we only divide by mass. What's your take on this? This is mini, but imagine if we do it 1M * 1000k whatever...

Reply:
very good idea, applied.